packages/dds/sequence/src/test/intervalundoredo.spec.ts

```diff
@@ -0,0 +1,395 @@
+		const end = sharedString.localReferencePositionToPosition(interval.end);
+		return { start, end };
+	});
+	assert.deepEqual(actualPos, expected, "intervals are not as expected");
+};
+
+// The code being tested currently does the wrong behavior. Currently,
+// the tests validate the bug and should be updated when the implementation
+// is fixed.
+describe.only("Undo/redo for interval collection operations", () => {
+	let sharedString: SharedString;
+	let dataStoreRuntime1: MockFluidDataStoreRuntime;
+	let containerRuntimeFactory: MockContainerRuntimeFactory;
+	let collection: IntervalCollection<SequenceInterval>;
+	let revertibles: SharedStringRevertible[];
+
+	beforeEach(() => {
+		dataStoreRuntime1 = new MockFluidDataStoreRuntime({ clientId: "1" });
+		sharedString = new SharedString(
+			dataStoreRuntime1,
+			"shared-string-1",
+			SharedStringFactory.Attributes,
+		);
+
+		containerRuntimeFactory = new MockContainerRuntimeFactory();
+
+		// Connect the first SharedString.
+		dataStoreRuntime1.local = false;
+		const containerRuntime1 = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime1);
+		const services1 = {
+			deltaConnection: containerRuntime1.createDeltaConnection(),
+			objectStorage: new MockStorage(),
+		};
+		sharedString.initializeLocal();
+		sharedString.connect(services1);
+		collection = sharedString.getIntervalCollection("test");
+		revertibles = [];
+	});
+
+	describe("with remote ops", () => {
+		let sharedString2: SharedString;
+
+		beforeEach(() => {
+			const dataStoreRuntime2 = new MockFluidDataStoreRuntime({ clientId: "2" });
+			const containerRuntime2 =
+				containerRuntimeFactory.createContainerRuntime(dataStoreRuntime2);
+			const services2 = {
+				deltaConnection: containerRuntime2.createDeltaConnection(),
+				objectStorage: new MockStorage(),
+			};
+
+			sharedString2 = new SharedString(
+				dataStoreRuntime2,
+				"shared-string-2",
+				SharedStringFactory.Attributes,
+			);
+			sharedString2.initializeLocal();
+			sharedString2.connect(services2);
+		});
+
+		it("handles remote remove of same range", () => {
+			sharedString.insertText(0, "hello world");
+
+			sharedString.on("sequenceDelta", (op) => {
+				if (op.opArgs.sequencedMessage === undefined) {
+					appendToMergeTreeDeltaRevertibles(sharedString, op.deltaArgs, revertibles);
+				}
+			});
+
+			collection.add(2, 4, IntervalType.SlideOnRemove);
+			containerRuntimeFactory.processAllMessages();
+
+			sharedString2.removeRange(0, 6);
+			sharedString.removeRange(0, 6);
+			assert.equal(sharedString.getText(), "world");
+			assert.equal(sharedString2.getText(), "world");
+			containerRuntimeFactory.processAllMessages();
+
+			assert.equal(revertibles.length, 1, "revertibles.length is not 1");
+			revertSharedStringRevertibles(sharedString, revertibles.splice(0));
+
+			assert.equal(sharedString.getText(), "hello world");
+			assertIntervals(sharedString, collection, [{ start: 2, end: 4 }]);
```

Comment:
It might be good to create a collection off of sharedString2 and check that the collection contains the expected intervals as well.